Source/platform/switch/keyboard.cpp

```diff
@@ -17,10 +17,25 @@ static void switch_keyboard_get(const char *guide_text, char *initial_text, int
 
 	if (R_SUCCEEDED(rc)) {
 		swkbdConfigMakePresetDefault(&kbd);
-		swkbdConfigSetGuideText(&kbd, guide_text);
-		swkbdConfigSetInitialText(&kbd, initial_text);
-		swkbdConfigSetStringLenMax(&kbd, max_len);
-		rc = swkbdShow(&kbd, buf, max_len);
+
+		// swkbConfigSetGuide/InitialText both copy the input string. They expect a null terminated c-string but we're
+		//  getting a string view which may be a substring of a larger std::string/c-string, so we copy to a temporary
+		//  null-terminated buffer first for safety
+		if (!guide_text.empty()) {
+			char buffer[guide_text.size() + 1];
+			memcpy(buffer, guide_text.data(), guide_text.size());
+			buffer[guide_text.size()] = '\0';
+			swkbdConfigSetGuideText(&kbd, buffer);
+		}
+		if (!initial_text.empty()) {
+			char buffer[initial_text.size() + 1];
+			memcpy(buffer, initial_text.data(), initial_text.size());
+			buffer[initial_text.size()] = '\0';
+			swkbdConfigSetInitialText(&kbd, buffer);
+		}

```

Comment:
This could potentially just use `std::string`?
```suggestion
		// swkbConfigSetGuide/InitialText both copy the input string. They expect a null terminated c-string but we're
		//  getting a string view which may be a substring of a larger std::string/c-string, so we copy to a temporary
		//  null-terminated string first for safety
		if (!guide_text.empty()) {
			std::string textCopy { guide_text };
			swkbdConfigSetGuideText(&kbd, textCopy.c_str());
		}
		if (!initial_text.empty()) {
			std::string textCopy { initial_text };
			swkbdConfigSetInitialText(&kbd, textCopy.c_str());
		}
```